src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -557,6 +556,20 @@ public static IResourceBuilder<ProjectResource> DisableForwardedHeaders(this IRe
         return builder;
     }
 
+    /// <summary>
+    /// Set a filter that determines if environment variables are injected for a given endpoint.
+    /// By default, all endpoints are included (if this method is not called).
+    /// </summary>
+    /// <param name="builder">The project resource builder.</param>
+    /// <param name="filter">The filter callback that returns true if and only if the endpoint should be included.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<ProjectResource> WithEndpointsInEnvironment(
+        this IResourceBuilder<ProjectResource> builder, Func<EndpointAnnotation, bool> filter)
+    {
+        builder.Resource.InjectEnvironmentForEndpoint = filter;
```

Comment:
Should we care about composition? Being able to get the previous filter?

Reply:
Pretty advanced scenario that no one will probably use, but I suppose we could. That could mean two things:
1. Automatically call the parent chain, transparently to the user. All of them need to return true for a given endpoint to be injected.
2. Pass the previous filter to the user and have them decide whether to delegate to it.

The second is a little more flexible as it lets you completely override previous filters. However, it makes the object model uglier for the user, e.g.

c#
```
.WithEndpointsInEnvironment(e,prev => e.Name != "FirstHttpEndpoint" && prev(e))
```

So if we do this, I'd vote for the first.

Reply:
Lets do it later.

Reply:
I ended up implementing the first approach (no change to user model). PTAL.